File: dmpw_linux_install_v5/code/api/API/DeMasterProCloud.DataAccess/Models/Holiday.cs
using System;
using System.Collections.Generic;

namespace DeMasterProCloud.DataAccess.Models
{
    public partial class Holiday
    {
        public int Id { get; set; }
        public int CompanyId { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime EndDate { get; set; }
        public string Name { get; set; }
        public bool Recursive { get; set; }
        public string Remarks { get; set; }
        public DateTime StartDate { get; set; }
        //public short Status { get; set; }
        public int Type { get; set; }
        public int UpdatedBy { get; set; }
        public DateTime UpdatedOn { get; set; }
        public bool IsDeleted { get; set; }

        public Company Company { get; set; }
    }
}

File: dmpw_linux_install_v7/code/webapp/src/types/holiday.ts
type EnumModel = {
  id: number;
  name?: string | null;
};

export type THoliday = {
  id: string;
  name?: string | null; // Holiday name
  companyId: number;
  type: number; // Holiday type
  startDate?: string | null; // Start date
  endDate?: string | null; // End date
  recursive: boolean;
  remarks?: string | null;
  holidayTypeItems?: EnumModel[] | null;
};
